File: scripts/test_phone.ts
/**
 * Quick smoke test for the Inkbox Phone TypeScript SDK.
 *
 * Loads INKBOX_API_KEY from ../.env and exercises every endpoint.
 */

import { readFileSync } from "fs";
import { resolve, dirname } from "path";
import { fileURLToPath } from "url";
import { InkboxPhone } from "../typescript/src/phone/index.js";

// Load .env from repo root
const __dirname = dirname(fileURLToPath(import.meta.url));
const envFile = resolve(__dirname, "../.env");
try {
  const contents = readFileSync(envFile, "utf-8");
  for (const line of contents.split("\n")) {
    const trimmed = line.trim();
    if (!trimmed || trimmed.startsWith("#") || !trimmed.includes("=")) continue;
    const idx = trimmed.indexOf("=");
    const key = trimmed.slice(0, idx).trim();
    const value = trimmed.slice(idx + 1).trim();
    if (!process.env[key]) process.env[key] = value;
  }
} catch {}

const API_KEY = process.env.INKBOX_API_KEY;
if (!API_KEY) {
  console.error("ERROR: INKBOX_API_KEY not set in .env or environment");
  process.exit(1);
}

const client = new InkboxPhone({ apiKey: API_KEY });

async function main() {
  // --- Numbers ---
  console.log("=== Listing phone numbers ===");
  const numbers = await client.numbers.list();
  for (const n of numbers) {
    console.log(`  ${n.number}  type=${n.type}  status=${n.status}  action=${n.incomingCallAction}`);
  }

  let number;
  if (numbers.length === 0) {
    console.log("\nNo numbers found. Provisioning a toll-free number...");
    number = await client.numbers.provision({ type: "toll_free" });
    console.log(`  Provisioned: ${number.number} (id=${number.id})`);
  } else {
    number = numbers[0];
    console.log(`\nUsing first number: ${number.number} (id=${number.id})`);
  }

  console.log("\n=== Get phone number ===");
  const fetched = await client.numbers.get(number.id);
  console.log(`  ${fetched.number}  pipeline_mode=${fetched.defaultPipelineMode}`);

  console.log("\n=== Update phone number ===");
  const updated = await client.numbers.update(number.id, {
    incomingCallAction: "auto_reject",
  });
  console.log(`  incomingCallAction=${updated.incomingCallAction}`);

  // --- Calls ---
  console.log("\n=== Listing calls ===");
  const calls = await client.calls.list(number.id, { limit: 5 });
  for (const c of calls) {
    console.log(`  ${c.id}  ${c.direction}  ${c.remotePhoneNumber}  status=${c.status}`);
  }

  if (calls.length > 0) {
    const call = calls[0];
    console.log(`\n=== Get call ${call.id} ===`);
    const detail = await client.calls.get(number.id, call.id);
    console.log(`  ${detail.direction}  ${detail.remotePhoneNumber}  pipeline=${detail.pipelineMode}`);

    // --- Transcripts (first call) ---
    console.log(`\n=== Listing transcripts for call ${call.id} ===`);
    const transcripts = await client.transcripts.list(number.id, call.id);
    for (const t of transcripts) {
      console.log(`  [${t.party}] seq=${t.seq} ts=${t.tsMs}ms: ${t.text.slice(0, 80)}`);
    }

    // --- Outbound call transcripts ---
    let outbound = calls.filter((c) => c.direction === "outbound");
    if (outbound.length === 0) {
      // Try fetching more calls to find an outbound one
      const allCalls = await client.calls.list(number.id, { limit: 200 });
      outbound = allCalls.filter((c) => c.direction === "outbound");
    }

    if (outbound.length > 0) {
      const ob = outbound[0];
      console.log(`\n=== Get outbound call ${ob.id} ===`);
      const obDetail = await client.calls.get(number.id, ob.id);
      console.log(`  ${obDetail.direction}  ${obDetail.remotePhoneNumber}  status=${obDetail.status}`);

      console.log(`\n=== Listing transcripts for outbound call ${ob.id} ===`);
      const obTranscripts = await client.transcripts.list(number.id, ob.id);
      if (obTranscripts.length > 0) {
        for (const t of obTranscripts) {
          console.log(`  [${t.party}] seq=${t.seq} ts=${t.tsMs}ms: ${t.text.slice(0, 80)}`);
        }
      } else {
        console.log("  (no transcripts)");
      }
    } else {
      console.log("\n  (no outbound calls found)");
    }
  }

  // --- Search ---
  console.log("\n=== Search transcripts ===");
  const results = await client.numbers.searchTranscripts(number.id, { q: "hello" });
  console.log(`  Found ${results.length} results`);
  for (const r of results.slice(0, 3)) {
    console.log(`  [${r.party}] ${r.text.slice(0, 80)}`);
  }

  // --- Webhooks ---
  console.log("\n=== Listing webhooks ===");
  const webhooks = await client.webhooks.list(number.id);
  for (const wh of webhooks) {
    console.log(`  ${wh.id}  url=${wh.url}  events=${wh.eventTypes}`);
  }

  console.log("\n=== Creating webhook ===");
  const hook = await client.webhooks.create(number.id, {
    url: "https://example.com/test-webhook",
    eventTypes: ["incoming_call"],
  });
  console.log(`  Created: ${hook.id}`);
  console.log(`  Secret: ${hook.secret}`);

  console.log("\n=== Updating webhook ===");
  const updatedHook = await client.webhooks.update(number.id, hook.id, {
    url: "https://example.com/updated-webhook",
  });
  console.log(`  Updated URL: ${updatedHook.url}`);

  console.log("\n=== Deleting webhook ===");
  await client.webhooks.delete(number.id, hook.id);
  console.log("  Deleted.");

  console.log("\nAll tests passed!");
}

main().catch((err) => {
  console.error("FAILED:", err);
  process.exit(1);
});

File: scripts/test_phone.py
"""
Quick smoke test for the Inkbox Phone Python SDK.

Loads INKBOX_API_KEY from ../.env and exercises every endpoint.
"""

import os
import sys
from pathlib import Path

from dotenv import load_dotenv
from inkbox.phone import InkboxPhone

load_dotenv(Path(__file__).resolve().parent.parent / ".env")

API_KEY = os.environ.get("INKBOX_API_KEY")
if not API_KEY:
    print("ERROR: INKBOX_API_KEY not set in .env or environment")
    sys.exit(1)

def main():
    with InkboxPhone(api_key=API_KEY) as client:

        # --- Numbers ---
        print("=== Listing phone numbers ===")
        numbers = client.numbers.list()
        for n in numbers:
            print(f"  {n.number}  type={n.type}  status={n.status}  action={n.incoming_call_action}")

        if not numbers:
            print("ERROR: No phone numbers found. Attach a number to your account first.")
            sys.exit(1)

        number = numbers[0]
        print(f"\nUsing first number: {number.number} (id={number.id})")

        print("\n=== Get phone number ===")
        fetched = client.numbers.get(number.id)
        print(f"  {fetched.number}  pipeline_mode={fetched.default_pipeline_mode}")

        print("\n=== Update phone number ===")
        updated = client.numbers.update(
            number.id,
            incoming_call_action="auto_reject",
        )
        print(f"  incoming_call_action={updated.incoming_call_action}")

        # --- Calls ---
        print("\n=== Listing calls ===")
        calls = client.calls.list(number.id, limit=5)
        for c in calls:
            print(f"  {c.id}  {c.direction}  {c.remote_phone_number}  status={c.status}")

        if calls:
            call = calls[0]
            print(f"\n=== Get call {call.id} ===")
            detail = client.calls.get(number.id, call.id)
            print(f"  {detail.direction}  {detail.remote_phone_number}  pipeline={detail.pipeline_mode}")

            # --- Transcripts (first call) ---
            print(f"\n=== Listing transcripts for call {call.id} ===")
            transcripts = client.transcripts.list(number.id, call.id)
            for t in transcripts:
                print(f"  [{t.party}] seq={t.seq} ts={t.ts_ms}ms: {t.text[:80]}")

            # --- Outbound call transcripts ---
            outbound = [c for c in calls if c.direction == "outbound"]
            if outbound:
                ob = outbound[0]
                print(f"\n=== Get outbound call {ob.id} ===")
                ob_detail = client.calls.get(number.id, ob.id)
                print(f"  {ob_detail.direction}  {ob_detail.remote_phone_number}  status={ob_detail.status}")

                print(f"\n=== Listing transcripts for outbound call {ob.id} ===")
                ob_transcripts = client.transcripts.list(number.id, ob.id)
                if ob_transcripts:
                    for t in ob_transcripts:
                        print(f"  [{t.party}] seq={t.seq} ts={t.ts_ms}ms: {t.text[:80]}")
                else:
                    print("  (no transcripts)")
            else:
                # Try fetching more calls to find an outbound one
                all_calls = client.calls.list(number.id, limit=200)
                outbound = [c for c in all_calls if c.direction == "outbound"]
                if outbound:
                    ob = outbound[0]
                    print(f"\n=== Get outbound call {ob.id} ===")
                    ob_detail = client.calls.get(number.id, ob.id)
                    print(f"  {ob_detail.direction}  {ob_detail.remote_phone_number}  status={ob_detail.status}")

                    print(f"\n=== Listing transcripts for outbound call {ob.id} ===")
                    ob_transcripts = client.transcripts.list(number.id, ob.id)
                    if ob_transcripts:
                        for t in ob_transcripts:
                            print(f"  [{t.party}] seq={t.seq} ts={t.ts_ms}ms: {t.text[:80]}")
                    else:
                        print("  (no transcripts)")
                else:
                    print("\n  (no outbound calls found)")

        # --- Search ---
        print("\n=== Search transcripts ===")
        results = client.numbers.search_transcripts(number.id, q="hello")
        print(f"  Found {len(results)} results")
        for r in results[:3]:
            print(f"  [{r.party}] {r.text[:80]}")

        # --- Webhooks ---
        print("\n=== Listing webhooks ===")
        webhooks = client.webhooks.list(number.id)
        for wh in webhooks:
            print(f"  {wh.id}  url={wh.url}  events={wh.event_types}")

        print("\n=== Creating webhook ===")
        hook = client.webhooks.create(
            number.id,
            url="https://example.com/test-webhook",
            event_types=["incoming_call"],
        )
        print(f"  Created: {hook.id}")
        print(f"  Secret: {hook.secret}")

        print("\n=== Updating webhook ===")
        updated_hook = client.webhooks.update(
            number.id,
            hook.id,
            url="https://example.com/updated-webhook",
        )
        print(f"  Updated URL: {updated_hook.url}")

        print("\n=== Deleting webhook ===")
        client.webhooks.delete(number.id, hook.id)
        print("  Deleted.")

        print("\nAll tests passed!")


if __name__ == "__main__":
    main()
